src/sqlfluff/rules/L069.py

```diff
@@ -29,147 +35,366 @@ class Rule_L069(BaseRule):
 
     **Anti-pattern**
 
-    Using CONVERT or :: instead of CAST when ``allow_non_standard_type_casting``
-    config is set to ``False`` (default).
+    Using mixture of CONVERT, :: and CAST when ``preferred_type_casting_style``
+    config is set to ``consistent`` (default).
 
     .. code-block:: sql
 
-        SELECT convert(varchar, 1) AS bar
-        FROM foo;
-
-        SELECT 1::varchar AS bar
+        SELECT
+            CONVERT(int, 1) AS bar,
+            100::int::text,
+            CAST(10 AS text) AS coo
         FROM foo;
 
     **Best practice**
 
-    Use ``CAST`` instead.
-    ``CAST`` is universally supported
+    Use consistent type casting style.
 
     .. code-block:: sql
 
-        SELECT CAST(1 AS varchar) AS bar,
+        SELECT
+            CAST(1 AS int) AS bar,
+            CAST(CAST(100 AS int) AS text),
+            CAST(10 AS text) AS coo
         FROM foo;
 
     """
 
     groups = ("all",)
-    config_keywords = ["allow_non_standard_type_casting"]
+    config_keywords = ["preferred_type_casting_style"]
     crawl_behaviour = SegmentSeekerCrawler({"function", "cast_expression"})
 
+    @staticmethod
+    def _get_children(segments: Segments) -> Segments:
+        return segments.children(
+            sp.and_(
+                sp.not_(sp.is_meta()),
+                sp.not_(
+                    sp.is_type(
+                        "start_bracket",
+                        "end_bracket",
+                        "whitespace",
+                        "newline",
+                        "casting_operator",
+                        "comma",
+                        "keyword",
+                    )
+                ),
+            )
+        )
+
+    @staticmethod
+    def _cast_fix_list(
+        context: RuleContext,
+        cast_arg_1: BaseSegment,
+        cast_arg_2: BaseSegment,
+        later_types=None,
```

Comment:
What is `later_types`? Could you add a type annotation to help readers of the code?

Reply:
Yes, sure.

Reply:
`later_types` is any other type after the first type e.g in `1::int::text`, text is the later type. I couldn't get a better naming.

Reply:
Type annotation added.